src/macros.rs

```diff
@@ -98,8 +122,24 @@ macro_rules! algos {
             }
         });
 
-        algos!(@algo xz ["xz"] XzDecoder XzEncoder<$inner> {
-            pub fn with_quality(inner: $inner, level: crate::Level) -> Self {
+        algos!(@algo-dec zstd ["zstd"] ZstdDecoder<$inner_dec> {
```

Comment:
Why can't you write 

```suggestion
impl<R> ZstdDecoder<R> {
```

then you won't need to change the macros.

Reply:
Bcs then we would need to specify the trait R implements
```rs
impl<R: tokio::io::AsyncBufRead> ZstdDecoder<R> {
``` 
which is dependent on the async executor.

Reply:
Thanks, you are right in this